File: Angular/practice/src/app/models/voter-election-candidate.ts
import { CandidateElection } from "./candidate-election";

export class VoterElectionCandidate {
    id:number;
    voterId:number;
    candidateElection:CandidateElection;
    electionId:number;

    constructor(candidateId:number, electionId:number, id:number,voteId:number,candidateElection:CandidateElection) {
        this.candidateElection = candidateElection;
        this.electionId = electionId; 
        this.id = id;
        this.voterId = voteId;
    }
}

File: spring-boot-voting-app/src/main/java/com/app/practice/pojos/VoterElectionCandidate.java
package com.app.practice.pojos;


import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name="voter_election_candidate")
@Getter
@Setter
@NoArgsConstructor
public class VoterElectionCandidate extends Base{

	
	
	public VoterElectionCandidate(Election election, Voter voter, CandidateElection candidateElection) {
		super();
		this.election = election;
		this.voter = voter;
		this.candidateElection = candidateElection;
	}

	@ManyToOne
	@JoinColumn(name="election_id")
	private Election election;
	
	@ManyToOne
	@JoinColumn(name="voter_id")
	private Voter voter;
	
	@ManyToOne
	@JoinColumn(name="candidate_election_id")
	private CandidateElection candidateElection;
}
